dv/views/api.py

```diff
@@ -1053,8 +1053,5 @@ def get_queryset(self):
                 # Django ORM generates an unnecessary complicated query here
             queryset = queryset.filter(q)
 
-        return queryset.select_related(
-            'project',
-        ).annotate(
-            total_allocation=Sum('allocation')
-        ).order_by('project_id').distinct()
+        project_list = queryset.values_list('project', flat=True).distinct()
+        return Project.objects.filter(code__in=project_list)
```

Comment:
You can change all filters above to work directly on `Project` model and you'll end up with a single query.